libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -390,14 +390,18 @@ private ServiceConnection obtainServiceConnection() {
     return new ServiceConnection() {
       @Override
       public void onServiceConnected(ComponentName className, IBinder service) {
-        TelemetryService.TelemetryBinder binder = (TelemetryService.TelemetryBinder) service;
-        telemetryService = binder.obtainService();
-        telemetryService.addServiceTask(MapboxTelemetry.this);
-        if (telemetryService.obtainBoundInstances() == 0) {
-          telemetryService.injectEventsQueue(queue);
+        if (service.getClass() == TelemetryService.TelemetryBinder.class) {
```

Comment:
Would these be equivalent to what you're doing on line 393: 
option A: `if (service instanceof TelemetryService.TelemetryBinder)`
option B: `TelemetryService.TelemetryBinder.class.isAssignableFrom(service.getClass())`
What are pros and cons of these approaches / performance implications?